src/tests/modify.test.js

```diff
@@ -0,0 +1,661 @@
+ * Thank you :)
+ * 🚧 🚧 🚧 🚧 🚧 🚧 🚧 🚧 🚧 🚧 🚧 🚧 🚧 🚧 🚧 🚧 🚧 🚧
+ */
+
+const schemaPet = {
+  type: 'object',
+  additionalProperties: false,
+  properties: {
+    has_pet: {
+      title: 'Has Pet',
+      description: 'Do you have a pet?',
+      oneOf: [
+        {
+          title: 'Yes',
+          const: 'yes',
+        },
+        {
+          title: 'No',
+          const: 'no',
+        },
+      ],
+      'x-jsf-presentation': {
+        inputType: 'radio',
+      },
+      type: 'string',
+    },
+    pet_name: {
+      title: "Pet's name",
+      description: "What's your pet's name?",
+      'x-jsf-presentation': {
+        inputType: 'text',
+      },
+      type: 'string',
+    },
+    pet_age: {
+      title: "Pet's age in months",
+      maximum: 24,
+      'x-jsf-presentation': {
+        inputType: 'number',
+      },
+      'x-jsf-errorMessage': {
+        maximum: 'Your pet cannot be older than 24 months.',
+      },
+      type: 'integer',
+    },
+    pet_fat: {
+      title: 'Pet fat percentage',
+      'x-jsf-presentation': {
+        inputType: 'number',
+        percentage: true,
+      },
+      type: 'integer',
+    },
+    pet_address: {
+      properties: {
+        street: {
+          title: 'Street',
+        },
+      },
+    },
+  },
+  required: ['has_pet'],
+  'x-jsf-order': ['has_pet', 'pet_name'],
+  allOf: [
+    {
+      id: 'pet_conditional_id',
+      if: {
+        properties: {
+          has_pet: {
+            const: 'yes',
+          },
+        },
+        required: ['has_pet'],
+      },
+      then: {
+        required: ['pet_name', 'pet_age', 'pet_fat'],
+      },
+      else: {
+        properties: {
+          pet_name: false,
+          pet_age: false,
+          pet_fat: false,
+        },
+      },
+    },
+  ],
+};
+
+const schemaAddress = {
+  properties: {
+    address: {
+      properties: {
+        street: {
+          title: 'Street',
+        },
+        number: {
+          title: 'Number',
+        },
+        city: {
+          title: 'City',
+        },
+      },
+    },
+  },
+};
+
+const imagineSomeBasicSchema = {
+  /* ... */
+};
+
+/*
+modify() Options:
+
+{
+  fields,
+  allFields,
+  order,
+  conditionals,
+  pick,
+  omit,
+  split, 
+}
+
+*/
+
+describe('modify() - attributes', () => {
+  it('replace base field', () => {
+    const result = modify(schemaPet, {
+      fields: {
+        pet_name: {
+          title: 'Your pet name',
+        },
+        // Q:❓[*1] Do you find it useful to be a callback function instead?
+        // You can access any raw attribute from the field to do whatever verification you need,
+        // but remember to be cautious, as the attrs value might change.
+        has_pet: (fieldAttrs) => {
+          const options = fieldAttrs.oneOf?.map(({ title }) => title).join(' or ') || '';
+          return {
+            title: 'Pet owner',
+            description: `Do you own a pet? ${options}?`, // "Do you own a pet? Yes or No?"
```

Comment:
Good question.

My first attempt is to say "Yes!" because whatever output you decide to give here, it's then consumed by the UI Library. 

But then there are exceptions that must be respected, such as the validators keywords like `minimum`, `maximum`, etc

This is a good challenge to balance the original JSON Schema types (a `title` being a string), and UI reactivity (a title being a UI component). I'll think about it and get back to you.

Reply:
My question was only about the `description` attribute. I don't think formik can handle the other fields being react components.

Reply:
"can handle the other fields being react components." — do you mean "other attributes", not "other fields", right? 

Formik is a headless component, so it can handle anything because the UI is not their direct concern. You as developer, build the legos that connect JSF to the UI Library (Formik). As long the `value` and validations (yup) are passed to Formik, everything else can customized as you/we wish.

Does this make sense?

Reply:
> "can handle the other fields being react components." — do you mean "other attributes", not "other fields", right?

Yes sorry 😅 

> Does this make sense?

Yeap It does! Thank you for the detailed explanation. So in that case, we only handle the `description` being a react component.